File: SingleLinkedList.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
    
    def append(self,data):
        newNode = Node(data)
        if(self.head is None):
            self.head = newNode
            return
        curr = self.head
        while(curr.next != None):
            curr = curr.next
        curr.next = newNode
    
    def delete(self,value):
        if self.head is None:
            print("List is empty, nothing to delete")
            return
        if self.head.data == value:
            self.head = self.head.next
            return
        curr = self.head
        while(curr.next is not None and curr.next.data != value):
            curr = curr.next
        if(curr.next is not None):
            curr.next = curr.next.next
        else:
            print("Value not found in the list")
    def display(self):
        curr = self.head
        while(curr is not None):
            print(curr.data,end="->")
            curr = curr.next
        print(None)


l = LinkedList()
l.append(3)
l.append(4)
l.display()
l.delete(4)
l.display()
        

File: SingleLinkedList.java
//Single Linked List
class Node<T> {
    T data;
    Node<T> next;

    public Node(T data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList<T> {
    Node<T> head;

    public LinkedList() {
        head = null;
    }

    public void append(T data) { // appends the data to the last node
        Node<T> newNode = new Node<>(data);
        if (head == null) {
            head = newNode;
            return;
        }
        Node<T> curr = head;
        while (curr.next != null) {
            curr = curr.next;
        }
        curr.next = newNode;
    }

    public void delete(T value) { // deletes the first occurance of the value
        if (head == null) {
            System.out.println("List is empty, nothing to delete");
            return;
        }

        if (head.data == value) {
            head = head.next;
            return;
        }

        Node<T> current = head;
        while (current.next != null && current.next.data != value) {
            current = current.next;
        }

        if (current.next != null) {
            current.next = current.next.next;
        } else {
            System.out.println("Value not found");
        }
    }

    public void display() {
        Node<T> curr = head;
        while (curr != null) {
            System.out.print(curr.data + "->");
            curr = curr.next;
        }
        System.out.println("null");
    }
}

public class SingleLinkedList {
    public static void main(String[] args) {
        LinkedList<Integer> l = new LinkedList<>();
        l.append(5);
        l.append(4);
        l.append(3);
        l.delete(3);
        l.display();
    }
}